File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] words = s.trim().split("\\s+");
        int start = 0;
        int end = words.length -1 ;
        while(start < end){
            String temp = words[start];
            words[start] = words[end];
            words[end] = temp;
            start++;
            end--;
        }

        String t = String.join(" ", words);

        return t;
    }
}

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        # words = s.split()
        # words.reverse()
        # return " ".join(words)


        words = s.split()
        result = []

        for i in range(len(words)-1, -1, -1):
            result.append(words[i])

        return " ".join(result)
        